### PR TITLE
Update documentation and setup.py

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

File: widgetsnbextension/setup.py [hunk 1]
```diff
@@ -215,9 +215,7 @@ def run(self):
 
 setuptools_args = {}
 install_requires = setuptools_args['install_requires'] = [
-    'notebook>=4.2.0b1',
-    # TODO: Add ipywidgets dependency when ipywidgets does not depend
-    # on widgetsnbextension.
+    'notebook>=4.2.0',
```
Comment (SylvainCorlay) [inline widgetsnbextension/setup.py]: Update notebook dependency.

File: widgetsnbextension/setup.py [hunk 2]
```diff
@@ -215,9 +215,7 @@ def run(self):
...
 ]
```
Comment (SylvainCorlay) [inline widgetsnbextension/setup.py]: Removed comment: (widgetsnbextension should be installed on a notebook server where there is no kernel (eventually). It does not make sense to make it depend on ipywidgets)

File: README.md [hunk 3]
```diff
@@ -37,7 +49,7 @@ or
 
 Then run
 
-    jupyter nbextension install --py --symlink widgetsnbextension
+    jupyter nbextension install --py widgetsnbextension
```
Comment (SylvainCorlay) [inline README.md]: symlink does not work on windows

File: docs/source/dev_install.md [hunk 4]
```diff
@@ -15,26 +15,29 @@ available in the PATH used with sudo.
...
         pip install -v -e .
         cd ..
+        jupyter nbextension install --py --symlink widgetsnbextension
+        jupyter nbextension enable --py widgetsnbextension
+
+3. Dev-install of the package (run from ipywidgets repo directory):
+
+        cd ipywidgets
+        pip install -v -e .
+
```
Comment (SylvainCorlay) [inline docs/source/dev_install.md]: we muse install widgetsnbextensions dev before ipywidgets for the pip version not to be picked up unnecessarily.

File: docs/source/dev_install.md [hunk 5]
```diff
@@ -15,26 +15,29 @@ available in the PATH used with sudo.
...
 
-3. Install the Jupyter Widgets nbextension
+2. Install the Jupyter Widgets nbextension
 
         cd widgetsnbextension
         npm install
         npm run update:widgets
         pip install -v -e .
         cd ..
+        jupyter nbextension install --py --symlink widgetsnbextension
```
Comment (SylvainCorlay) [inline docs/source/dev_install.md]: we were missing the nbextension install here.

File: setup.py [hunk 6]
```diff
@@ -106,10 +106,11 @@
...
     'traitlets>=4.2.0',
-    'widgetsnbextension>=1.0.0b4',   # TODO: Remove this dependency in 6.0
-                                     # (We should be able to install the python
-                                     #  side of ipywidgets on bare python
-                                     #  kernels)
+    # TODO: Dynamically add this dependency
+    # only if notebook 4.x is installed in this
+    # interpreter, to allow ipywidgets to be
+    # installed on bare kernels.
+    'widgetsnbextension>=1.0.0b4',
```
Comment (SylvainCorlay) [inline setup.py]: So a path forward for the dependency of ipywidgets to widgetsnbextension is the following.
 - you can install ipywidgets on a bare kernel. If notebook 4.x is not there, don't add the widgetsnbextension.
 - If notebook 4.x is installed on this interpreter, it does not hurt to install widgetsnbextension and we still keep the one-step install for standalone users.

File: widgetsnbextension/setup.py (hunk 1 above)
Comment (jdfreder) [inline widgetsnbextension/setup.py]: 🎉

File: widgetsnbextension/setup.py (hunk 2 above)
Comment (jdfreder) [inline widgetsnbextension/setup.py]: 👍

File: setup.py (hunk 6 above)
Comment (jdfreder) [inline setup.py]: Sounds good!

**Pull request merged**